### PR TITLE
Add option to select building GeoReferencer plugin

### PR DESCRIPTION
**Pull request by cjmayo**
The decision to build the GeoReferencer plugin is based on whether GSL is already installed.
Such "automagic" dependencies are a problem for Gentoo Linux because they can result in the package manager not being able to  track the links between packages. 

The first version of this PR preserves the current behaviour by default, but allows the installer to prevent the plugin being built with -DWITH_GEOREFERENCER=OFF.

Ideally REQUIRED would be added to FIND_PACKAGE(GSL) to check GSL is installed but this would mean a change to the current behaviour, probably changing the default to WITH_GEOREFERENCER OFF.

There is already a precedent with the Globe plugin that also uses FIND_PACKAGE with REQUIRED for external libraries and has a WITH_GLOBE option.

For Globe the FIND_PACKAGE REQUIRED is duplicated in the top-level CMakeLists.txt, presumably to create an earlier failure rather than a long way into compilation?

I'm happy to develop this further if it might be accepted. It would help to have some pointers about the desired behaviour and documentation updates required.

### CODE REVIEW

**Comment by m-kuhn** [comment]
I agree this kind of magic is not desirable. Thank you for looking into it.

I would say, make it a required dependency which is listed only in the georeferencer plugins CMakeLists.txt  (this will not affect build times but make it less prone to have leftovers if we ever remove georeferencer in favor of something else) and add a new parameter `-DWITH_GEOREFERENCER` defaulting to true (some people will have to adjust but this will force an easily fixable conscious decision).

**Comment by cjmayo** [comment]
OK. Rebased on master and made GSL required in the plugin CMakeLists.txt.

If the option was selected but GSL was not installed the build was continuing:
```
-- Found QwtPolar: /usr/lib64/libqwtpolar.so
FindGSL.cmake: gsl-config not found. Please set it manually. GSL_CONFIG=GSL_CONFIG-NOTFOUND
-- txt2tags not found - disabled
```
Therefore added a FATAL_ERROR to cmake/FindGSL.cmake.

This is causing Travis to fail - it doesn't have GSL?:
https://travis-ci.org/qgis/QGIS/jobs/205319643

FindGSL.cmake is now bundled with CMake, so ideally I would have deleted cmake/FindGSL.cmake.
https://cmake.org/cmake/help/v3.7/module/FindGSL.html
But this was introduced in CMake 3.2:
https://cmake.org/cmake/help/v3.7/release/3.2.html
I don't know if raising the minimum CMake would be acceptable (could be used to replace other CMake Modules). 

I see Debian stable has less than 3.2 without using backports.
https://packages.debian.org/search?keywords=cmake&searchon=names&exact=1&suite=all&section=all

**Comment by m-kuhn** [comment]
Thanks.

Can you add `gsl` (or `gsl-devel`?) in apt-packages in `.travis.yml` (in the first section (main build))?

I'd just leave the `FindGSL.cmake` in place for now if in doubt.

**Comment by cjmayo** [comment]
I went with `libgsl0-dev`, seems to work.

File: cmake/FindGSL.cmake [hunk 1]
```diff
@@ -118,7 +118,7 @@ ELSE(WIN32)
         )
       
       ELSE(GSL_CONFIG)
-        MESSAGE("FindGSL.cmake: gsl-config not found. Please set it manually. GSL_CONFIG=${GSL_CONFIG}")
+        MESSAGE(FATAL_ERROR "FindGSL.cmake: gsl-config not found. Please set it manually. GSL_CONFIG=${GSL_CONFIG}")
```
Comment (m-kuhn) [inline cmake/FindGSL.cmake]: Could this be made a bit more verbose? And what I also wonder is the use of `${GSL_CONFIG}` in here, after all it's in the else branch where it has not been found.

> FindGSL.cmake: gsl-config not found. Please install the `libgsl` development libraries or set the path with `-DGSL_CONFIG=/path/to/gsl-config`.

(I know it wasn't you who wrote it ;) )

Comment (cjmayo) [inline cmake/FindGSL.cmake]: Message reworded as proposed.

**Comment by m-kuhn** [comment]
Thanks a lot!

**Pull request merged**